From cadc63dd0f01d600d2b25ec4405982a85d88a333 Mon Sep 17 00:00:00 2001
From: Grygorii Strashko <grygorii.strashko@ti.com>
Date: Thu, 11 Feb 2021 01:05:13 +0200
Subject: [PATCH 10/15] phc2sys: ti: fix extts for "phc2sys.c always kill servo
 when switch to master for BC"

Update for extts support.

Signed-off-by: Grygorii Strashko <grygorii.strashko@ti.com>
---
 phc2sys.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/phc2sys.c b/phc2sys.c
index fc7760d..ab53992 100644
--- a/phc2sys.c
+++ b/phc2sys.c
@@ -1186,10 +1186,10 @@ static int autocfg_extts_init_clocks(struct node *node,
 			continue;
 
 		autocfg_extts_clock_settime(node, node->master, clock);
-
-		pr_debug("reset %s servo", clock->device);
-		servo_reset(clock->servo);
-		clock->servo_state = SERVO_UNLOCKED;
+		if (clock->servo) {
+			servo_reset(clock->servo);
+			clock->servo_state = SERVO_UNLOCKED;
+		}
 
 		if (clock->offset_stats) {
 			stats_reset(clock->offset_stats);
-- 
2.17.1

